procrastinate/aiopg_connector.py

```diff
@@ -20,18 +20,37 @@
 
 def wrap_exceptions(coro: CoroutineFunction) -> CoroutineFunction:
     """
-    Wrap psycopg2 and aiopg errors as connector exceptions
-    This decorator is expected to be used on coroutine functions only
+    Wrap psycopg2 and aiopg errors as connector exceptions. In case we get a
+    psycopg2.OperationalError with a "server closed the connection
+    unexpectedly" message we retry a number of times.
+
+    This decorator is expected to be used on coroutine functions only.
     """
 
     @functools.wraps(coro)
     async def wrapped(*args, **kwargs):
-        try:
-            return await coro(*args, **kwargs)
-        except psycopg2.errors.UniqueViolation as exc:
-            raise exceptions.UniqueViolation(constraint_name=exc.diag.constraint_name)
-        except psycopg2.Error as exc:
-            raise exceptions.ConnectorException from exc
+        cnt = 0
+        max_tries = (
```

Comment:
This deserves the hell of a comment :D
- Why are we doing this maxsize times ?
- Why are we using a while loop and not a classic for loop ?

Also, we could use an [`itertools.count()`](https://docs.python.org/3.8/library/itertools.html#itertools.count)

And finally, I think we mentionned we wanted to put this in a differentdecorator ? This would help wrap_exceptions not to become too big.

Reply:
> And finally, I think we mentionned we wanted to put this in a differentdecorator ? This would help wrap_exceptions not to become too big.

Yep, but this led to code duplication, and it was not so clear where to use `wrap_exceptions` and `wrap_query_exceptions`.

Reply:
Why was there code duplication ?

I was imagining the query methods be decorated by the 2 decorators, if this is unclear.

Reply:
Right, I hadn't realized that you imagined two decorations for the query functions. That should work too!

Reply:
I added a commit that adds a `wrap_query_exceptions` decorator. I don't find this better than what my first commit provides, but maybe you had something else in mind.

Reply:
> This deserves the hell of a comment :D

You're right. I'll add them.

Reply:
> I don't find this better than what my first commit provides, but maybe you had something else in mind.

Well you have split a complex function into 2 less complex more scoped functions. I think it's overally better.